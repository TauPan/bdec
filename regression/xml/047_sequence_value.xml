<protocol>
    <sequence name="a">
        <reference name="two digits" />
        <field name="buffer" length="(${two digits} * 8)" type="text" />
    </sequence>
    <common>
        <sequence name="digit" value="(${text digit} - 6 * 8)">
            <field name="text digit" length="8" type="integer" min="48" max="57" />
        </sequence>
        <sequence name="two digits" value="((${digit 1.digit} * 10) + ${digit 2.digit})">
            <sequence name="digit 1">
                <reference name="digit" />
            </sequence>
            <sequence name="digit 2">
                <reference name="digit" />
            </sequence>
        </sequence>
    </common>
</protocol>
