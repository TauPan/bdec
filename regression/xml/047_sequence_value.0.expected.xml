<a>
            <two-digits>
              <digit-1><digit><text-digit>50</text-digit></digit></digit-1>
              <digit-2><digit><text-digit>49</text-digit></digit></digit-2>
            </two-digits>
            <buffer>xxxxxxxxxxxxxxxxxxxxx</buffer>
          </a>